KPI/views/firefox_desktop_usage_forecast.view.lkml

```diff
@@ -0,0 +1,301 @@
+      {% condition firefox_desktop_usage_2021.attributed %} (attributed) {% endcondition %}
+      """, r"(\(.*\))"), '');
+
+    DECLARE row_count INT64;
+
+    SET row_count = (
+      SELECT COUNT(*)
+      FROM mozdata.analysis.dou_forecasts
+      WHERE key = current_key);
+
+    IF row_count = 0 THEN
+      CREATE OR REPLACE MODEL
+      ${SQL_TABLE_NAME} OPTIONS(
+        model_type='ARIMA',
+        time_series_data_col='new_profiles',
+        time_series_timestamp_col='submission_date'
+      ) AS
+      SELECT
+        submission_date,
+        new_profiles
+      FROM
+        ${firefox_desktop_usage_2021.SQL_TABLE_NAME}
+      WHERE
+        submission_date < '2021-01-19';
+    END IF; ;;
+  }
+}
+
+view: insert_stmnt {
+  derived_table: {
+    sql_trigger_value: current_timestamp() ;;
+    sql_create:
+      DECLARE current_key STRING DEFAULT ARRAY_TO_STRING(REGEXP_EXTRACT_ALL("""
+      {% condition firefox_desktop_usage_2021.activity_segment %} activity_segment {% endcondition %}
+      {% condition firefox_desktop_usage_2021.campaign %} campaign {% endcondition %}
+      {% condition firefox_desktop_usage_2021.channel %} channel {% endcondition %}
+      {% condition firefox_desktop_usage_2021.content %} content {% endcondition %}
+      {% condition firefox_desktop_usage_2021.country %} country {% endcondition %}
+      {% condition firefox_desktop_usage_2021.distribution_id %} distribution_id {% endcondition %}
+      {% condition firefox_desktop_usage_2021.id_bucket %} id_bucket {% endcondition %}
+      {% condition firefox_desktop_usage_2021.medium %} medium {% endcondition %}
+      {% condition firefox_desktop_usage_2021.os %} os {% endcondition %}
+      {% condition firefox_desktop_usage_2021.source %} source {% endcondition %}
+      {% condition firefox_desktop_usage_2021.attributed %} (attributed) {% endcondition %}
+      """, r"(\(.*\))"), '');
+
+      DECLARE row_count INT64;
+
+      SET row_count = (
+        SELECT COUNT(*)
+        FROM mozdata.analysis.dou_forecasts
+        WHERE key = current_key);
+
+      IF row_count = 0 THEN
+
+      INSERT INTO mozdata.analysis.dou_forecasts_cache
+      WITH historic_data AS (
+        SELECT
+          CAST(submission_date AS TIMESTAMP) AS submission_date,
+
+          CAST(dau AS FLOAT64) AS dau_forecast,
+          CAST(NULL AS FLOAT64) AS dau_target,
+          CAST(NULL AS FLOAT64) AS dau_forecast_lower,
+          CAST(NULL AS FLOAT64) AS dau_forecast_upper,
+
+          CAST(new_profiles AS FLOAT64) AS new_profiles_forecast,
+          CAST(NULL AS FLOAT64) AS new_profiles_target,
+          CAST(NULL AS FLOAT64) AS new_profiles_forecast_lower,
+          CAST(NULL AS FLOAT64) AS new_profiles_forecast_upper,
+        FROM
+          ${firefox_desktop_usage_2021.SQL_TABLE_NAME}
+        WHERE
+          submission_date BETWEEN '2021-01-01' AND '2021-01-18'
+      ), predicted_data AS (
```

Comment:
Okay, something happened here. Basically the BQ ML API only gives results for _after_ the last date of your training set. So in this case our training set goes to `2021-01-18`, which caused our forecasted CDOU to be artificially low, since dou and cdou for 01/01-01/18 are forecasted 0.

I deal with this by putting the actual dau in for those dates, which gives us appropriate values for cdou 2021 (exact same up to 2021-01-18). However the plots are odd with a straight line at the beginning.